### PR TITLE
add write_plugin_graph test

### PR DESCRIPTION
**Pull request by tjhei**
This adds a test for #1777

Maybe this is not worth merging as is, because the output will break every time a new plugin is added, so we should add the feature to only output selected plugins first?

### CODE REVIEW

**Comment by bangerth** [comment]
Yes, I'm not sure this is a good idea either. How often do we add new plugins?

**Comment by bangerth** [comment]
Apparently the order in which plugins are output is also not deterministic from one machine to the next :-(

**Comment by gassmoeller** [comment]
I would guess about once per month, of course much more often during hackathons. Maybe we could add the test without the actual graph output? This would at least test that the function does not crash. Otherwise I am not really in favor of having a test that needs to be updated every so often.

**Comment by tjhei** [comment]
> Apparently the order in which plugins are output is also not deterministic from one machine to the next

Initialization order of static members, I assume?

> Otherwise I am not really in favor of having a test that needs to be updated every so often.

How about the current solution? I am checking that some substrings appear in the text.

**Comment by bangerth** [comment]
Definitely initialization order of static variables.

File: tests/write_plugin_graph/screen-output [hunk 1]
```diff
@@ -0,0 +1,28 @@
+-----------------------------------------------------------------------------
+-----------------------------------------------------------------------------
+
+Loading shared library <./libwrite_plugin_graph.so>
+
+Number of active cells: 16 (on 3 levels)
+Number of degrees of freedom: 268 (162+25+81)
+
+found digraph Plugins
+found label="TestMaterial"
```
Comment (bangerth) [inline tests/write_plugin_graph/screen-output]: Ah, clever -- I was wondering why you are creating the test material model. Nice idea!

**Comment by bangerth** [comment]
I like it!

As a complete aside, you could have made the test slightly simpler if you stored the return value of `signals.post_constraints_creation.connect`, which is a handle to the connection. You could then in the function that produces the output just release the connection and wouldn't have to track whether this is the `first` time you get into the function.

**Pull request merged**